TS2315 when using `@template` more than once in a file

## Steps to reproduce

Create the following files:

```jsonc
// tsconfig.json
{
  "compilerOptions": {
    "allowJs": true,
    "checkJs": true,
    "strict": false,

    "declaration": true,
    "emitDeclarationOnly": true,
    "outDir": "types",

    "target": "ES2022",
    "module": "NodeNext",
    "moduleResolution": "NodeNext",
    "lib": ["ES2023"],

    "skipLibCheck": true,
    "esModuleInterop": true
  },
  "include": ["*.js"]
}
````

```js
// set.js
/**
 * @template D
 * @typedef {(a: D, b: D) => number} SomeUnrelatedGeneric
 */

/**
 * @template T
 */
class SomeGenericClass {
  /**
   * @param {Set<T>} param
   */
  constructor(param) {}
}

module.exports = SomeGenericClass;
```

```js
// index.js
const Set = require("./set");

/** @typedef {Set<string>} MyDefinedType */
```

## Behavior with `typescript@6.0`

The declaration output is generated successfully.

## Behavior with `tsgo`

`tsgo` reports the following error:

```txt
index.js:3:15 - error TS2315: Type 'export=' is not generic.

3 /** @typedef {Set<string>} MyDefinedType */
                ~~~~~~~~~~~

Found 1 error in index.js:3
```

Removing the unrelated generic typedef from `set.js` fixes the issue:

```js
/**
 * @template D
 * @typedef {(a: D, b: D) => number} SomeUnrelatedGeneric
 */
```

--- a/internal/checker/checker.go
+++ b/internal/checker/checker.go
@@ -15691,8 +15691,9 @@ func (c *Checker) resolveEntityName(name *ast.Node, meaning ast.SymbolFlags, ign
 			(symbol.Flags&ast.SymbolFlagsAlias != 0 || name.Parent != nil && name.Parent.Kind == ast.KindExportAssignment) {
 			c.markSymbolOfAliasDeclarationIfTypeOnly(getAliasDeclarationFromName(name), nil)
 		}
-		if symbol.Flags&meaning == 0 && !dontResolveAlias && symbol.Flags&ast.SymbolFlagsAlias != 0 {
-			return c.resolveAlias(symbol)
+		// We know a symbol with the given meaning exists along the alias chain, so resolve until we find it.
+		for symbol.Flags&meaning == 0 && !dontResolveAlias && symbol.Flags&ast.SymbolFlagsAlias != 0 {
+			symbol = c.resolveAlias(symbol)
 		}
 	}
 	return symbol

--- a/testdata/baselines/reference/compiler/cjsExportGenericTypes.symbols
+++ b/testdata/baselines/reference/compiler/cjsExportGenericTypes.symbols
@@ -1,0 +1,35 @@
+//// [tests/cases/compiler/cjsExportGenericTypes.ts] ////
+
+=== index.js ===
+const Set = require("./set");
+>Set : Symbol(Set, Decl(index.js, 0, 5))
+>require : Symbol(require)
+>"./set" : Symbol("./set", Decl(set.js, 0, 0))
+
+/** @typedef {Set<string>} MyDefinedType */
+
+=== set.js ===
+/**
+ * @template D
+ * @typedef {(a: D, b: D) => number} SomeUnrelatedGeneric
+ */
+
+/**
+ * @template T
+ */
+class SomeGenericClass {
+>SomeGenericClass : Symbol(SomeGenericClass, Decl(set.js, 0, 0))
+
+  /**
+   * @param {Set<T>} param
+   */
+  constructor(param) {}
+>param : Symbol(param, Decl(set.js, 12, 14))
+}
+
+module.exports = SomeGenericClass;
+>module.exports : Symbol(exports, Decl(set.js, 0, 0))
+>module : Symbol(module, Decl(set.js, 0, 0))
+>exports : Symbol(exports, Decl(set.js, 0, 0))
+>SomeGenericClass : Symbol(SomeGenericClass, Decl(set.js, 0, 0))
+

--- a/testdata/baselines/reference/compiler/cjsExportGenericTypes.types
+++ b/testdata/baselines/reference/compiler/cjsExportGenericTypes.types
@@ -1,0 +1,37 @@
+//// [tests/cases/compiler/cjsExportGenericTypes.ts] ////
+
+=== index.js ===
+const Set = require("./set");
+>Set : typeof SomeGenericClass
+>require("./set") : typeof SomeGenericClass
+>require : any
+>"./set" : "./set"
+
+/** @typedef {Set<string>} MyDefinedType */
+
+=== set.js ===
+/**
+ * @template D
+ * @typedef {(a: D, b: D) => number} SomeUnrelatedGeneric
+ */
+
+/**
+ * @template T
+ */
+class SomeGenericClass {
+>SomeGenericClass : SomeGenericClass<T>
+
+  /**
+   * @param {Set<T>} param
+   */
+  constructor(param) {}
+>param : Set<T>
+}
+
+module.exports = SomeGenericClass;
+>module.exports = SomeGenericClass : typeof SomeGenericClass
+>module.exports : typeof SomeGenericClass
+>module : { exports: typeof SomeGenericClass; }
+>exports : typeof SomeGenericClass
+>SomeGenericClass : typeof SomeGenericClass
+

--- a/testdata/tests/cases/compiler/cjsExportGenericTypes.ts
+++ b/testdata/tests/cases/compiler/cjsExportGenericTypes.ts
@@ -1,0 +1,25 @@
+// @checkJs: true
+// @noEmit: true
+
+// @filename: set.js
+/**
+ * @template D
+ * @typedef {(a: D, b: D) => number} SomeUnrelatedGeneric
+ */
+
+/**
+ * @template T
+ */
+class SomeGenericClass {
+  /**
+   * @param {Set<T>} param
+   */
+  constructor(param) {}
+}
+
+module.exports = SomeGenericClass;
+
+// @filename: index.js
+const Set = require("./set");
+
+/** @typedef {Set<string>} MyDefinedType */

